File: im/프로그래머스/프로그래머스 Lv2/멀리뛰기.java
class Solution {

    public long solution(int n) {
        if(n == 1) return 1;
        if(n == 2) return 2;
        long answer = 0;

        long f1 = 1;
        long f2 = 2;

        for( int i = 3; i <= n; i++){
            answer = f1 + f2;
            f1 = f2;
            f2 = answer % 1234567;
        }


        return answer % 1234567;
    }
}

File: baik/Algorithm/lv2/멀리뛰기.swift
//
//  멀리뛰기.swift
//  Algorithm
//
//  Created by jonghyun baik on 6/5/24.
//

import Foundation

func solution(_ n:Int) -> Int {
  var fibonacci: [Int] = [0, 1]
  
  for i in 2...n {
    fibonacci.append((fibonacci[i-2] + fibonacci[i-1]) % 1234567)
  }
  
  return fibonacci[n]
}
